packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -613,27 +615,49 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             combinedRuntimeOptions,
             containerScope,
             logger,
+            existing,
             requestHandler,
             storage);
 
         if (combinedRuntimeOptions.addGlobalAgentSchedulerAndLeaderElection !== false) {
             // Create all internal data stores if not already existing on storage or loaded a detached
             // container from snapshot(ex. draft mode).
-            if (!context.existing) {
+            if (!existing) {
                 await runtime.createRootDataStore(AgentSchedulerFactory.type, agentSchedulerId);
             }
         }
 
         return runtime;
     }
 
-    public get id(): string {
-        return this.context.id;
+    /**
+     * Load the stores from a snapshot and returns the runtime.
+     *
+     * @deprecated Please use ContainerRuntime.loadStateful
+     *
+     * @param context - Context of the container.
+     * @param registry - Mapping to the stores.
+     * @param requestHandlers - Request handlers for the container runtime
+     * @param runtimeOptions - Additional options to be passed to the runtime
+     */
+    public static async load(
+        context: IContainerContext,
+        registryEntries: NamedFluidDataStoreRegistryEntries,
+        requestHandler?: (request: IRequest, runtime: IContainerRuntime) => Promise<IResponse>,
+        runtimeOptions?: IContainerRuntimeOptions,
+        containerScope: IFluidObject = context.scope,
+    ): Promise<ContainerRuntime> {
+        return this.loadStateful(
```

Comment:
I'd probably keep the name, i.e. just add existing argument to existing load() function.
Implementation of container runtime and runtime factory are within same compat layer, so breaking API is not a big deal (though in future we might indeed want to go the route you are taking with never changing / deleting API before unless it was deprecated first).
Whoever picks up new version of FF would need to adjust their code, at minimum - passing context.existing to Container.load(), but this will also force them to look around and possibly to move to new paradigm